src/cache_test.ts

```diff
@@ -143,11 +144,11 @@ describe('ListWatchCache', () => {
         expect(pathOut).to.equal('/some/path');
         expect(cache.list()).to.deep.equal(list);
 
-        expect(cache.get('name1')).to.equal(list[0]);
-        expect(cache.get('name2')).to.equal(list[1]);
+        expect(cache.get('name1', 'default')).to.equal(list[0]);
+        expect(cache.get('name2', 'default')).to.equal(list[1]);
 
         expect(cache.list('default')).to.deep.equal(list);
-        expect(cache.list('non-existent')).to.be.undefined;
+        expect(cache.list('non-existent')).to.deep.equal([]);
```

Comment:
The test and the type of the `list` return disagreed, I went with returning an empty array if its not found. I can leave the diagreement in place or update the types an undefined is preferred to be returned.

Reply:
Empty array seems fine.